packages/matter.js/src/protocol/interaction/InteractionClient.ts

```diff
@@ -35,9 +36,78 @@ export interface AttributeStatus {
+    private readonly attributeToId = <{ [key: number]: string }>{};
+    readonly commands = <{ [P in keyof C]: SignatureFromCommandSpec<C[P]> }>{};
+    private interactionClient: InteractionClient | undefined;
+
+    constructor(
+        clusterDef: Cluster<any, A, C, any>,
+        readonly endpointId: number,
+        interactionClient?: InteractionClient,
+    ) {
+        const { id: clusterId, name, commands, attributes } = clusterDef;
+        this.id = clusterId;
+        this.name = name;
+
+        // Add accessors
+        for (const attributeName in attributes) {
+            const attribute = attributes[attributeName];
+            (this.attributes as any)[attributeName] = new AttributeClient(attribute, attributeName, endpointId, clusterId);
+            this.attributeToId[attribute.id] = attributeName;
+        }
+
+        if (interactionClient !== undefined) {
+            this.bindToInteractionClient(interactionClient);
+        }
+
+        // Add command calls
+        for (const commandName in commands) {
+            const { requestId, requestSchema, responseId, responseSchema, optional } = commands[commandName];
+
+            (this.commands as any)[commandName] = async <RequestT, ResponseT>(request: RequestT) => {
+                if (this.interactionClient === undefined) {
+                    throw new Error("InteractionClient not set");
+                }
+                return this.interactionClient.invoke<Command<RequestT, ResponseT>>(endpointId, clusterId, request, requestId, requestSchema, responseId, responseSchema, optional);
+            };
+        }
+    }
+
+    bindToInteractionClient(interactionClient: InteractionClient) {
+        this.interactionClient = interactionClient;
+        for (const attributeName in this.attributes) {
+            (this.attributes as any)[attributeName].bindToInteractionClient(this.interactionClient);
+        }
+    }
+
+    subscribe(minIntervalFloorSeconds: number, maxIntervalCeilingSeconds: number) {
+        if (this.interactionClient === undefined) {
+            throw new Error("InteractionClient not set");
+        }
+        return this.interactionClient.subscribeMultipleAttributes([{ endpointId: this.endpointId, clusterId: this.id }], minIntervalFloorSeconds, maxIntervalCeilingSeconds, (data) => {
+            const { path, value } = data;
+            const attributeName = this.attributeToId[path.attributeId];
+            if (attributeName === undefined) {
+                logger.warn("Unknown attribute id", path.attributeId);
+                return;
+            }
+            (this.attributes as any)[attributeName].update(value);
+        });
+    }
+}
+
+export function ClusterClientInstance<CommandT extends Commands, AttributeT extends Attributes>(interactionClient: InteractionClient, endpointId: number, clusterDef: Cluster<any, AttributeT, CommandT, any>): ClusterClientInstance<CommandT, AttributeT> {
     const result: any = {};
```

Comment:
```javaascript
const { id: clusterId, name, commands, attributes } = clusterDef;
const result: any = { id: clusterId, name, endpointId, attributes, commands };
```

Reply:
Yes, this ClientInstance is currentl kind iof unused and will coming in soon again. WIll then consider your idea. thank you very much

Reply:
Really looking forward to a Client Cluster for things like sensors.

Reply:
on my plan for this weekend

Reply:
PS: Ok CLusterClients on a device (and not controller) might be another topic ... I assume that then also events join the game and we do not have them till now implemented

Reply:
I wonder if it can be done without all the events implemented just yet,
Like just display a static value, like an information dummy card for now.
I really wanna see what that looks like on the home app.

Reply:
Ok are we talkling about the same? If you want to offer a TemperatureSensorDevice this work already because there a TemperatureServer is used and NOT a client. so what exactly you want to do?

Reply:
I thought that a device client is needed in order to report information about a device to the controller.
Like a server is something that listens to commands, while a client is something that sends commands back
to the matter controller and does not accept commands, as that would make it a device server.
You're saying that a temperature sensor is a device server?

Reply:
Another example of a device client would be a physical switch, since its a state setter, it does not react to anything, its the thing that other things react to it. (until you add a state indicator and it becomes both a server and a client...).

Reply:
>  You're saying that a temperature sensor is a device server?

yes https://github.com/project-chip/matter.js/blob/main/packages/matter.js/src/common/DeviceTypes.ts#L630-L643

For cluster Clients this is a good example:

https://github.com/project-chip/matter.js/blob/main/packages/matter.js/src/common/DeviceTypes.ts#L690-L714

For ClusterClients, when I understand the specs rught, you utilize "Binding" as feature which uses the Binding cluster (see 9.4 and 9.6 in core specs).

So as I understand that is on example of an onOff Device:
OnOff device can have an optional Occupancy sensor client. This menas that you can optionally bind the onoff device to an Occupancysensor device (which then have a Occupancy Cluster SERVER). The the light can e.g. subscribe to that occupancy sensor and turn on/off the light based on this sensor.

So if you want to have a "device that measures a teperature and provides that value to ios Home app" then you need a Temperature devcie with a TemperatureMeasurement ClusterServer.

I hope now it got more clearer?